### PR TITLE
UISchema API docs

### PR DESCRIPTION
**Pull request by brentschaus-okta**
## Description:
- **What's changed?** Adds reference docs for UISchema API
- **Is this PR related to a Monolith release?** No.

### Resolves:

* [OKTA-448404](https://oktainc.atlassian.net/browse/OKTA-448404)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 1]
```diff
@@ -0,0 +1,750 @@
+---
+title: UI Schema API
+category: management
+---
+
+# UI Schema API
+
+<ApiLifecycle access="ie" />
+
+The Okta UI Schema API allows you to control how inputs appear on a form.
```
Comment (brentschaus-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:10]: I think that we need to clarify which forms they can update.

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 2]
```diff
@@ -0,0 +1,748 @@
...
+```
+
+### UI Schema Object
+
+#### UI Schema properties
+
+| Property      | Type   | Description | Supported Values |
+| ------------- | ------ | ----------- | ---------------- |
+| `buttonLabel` | String | Specifies the button label for the `submit` button at the bottom of the enrollment form. Defaults to `Submit`. |
+| `elements`    | Array  |  |
```
Comment (brentschaus-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:686]: @jakebacher2-okta There are a few properties in the "UI Schema object" section that are missing descriptions. They still had the template placeholder content which I deleted.

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 3]
```diff
@@ -0,0 +1,748 @@
...
+### UI Schema Element object
+
+The UI Schema Element object specifies the configuration of an input field on an enrollment form.
+
+#### UI Schema Element properties
+
+| Property | Type | Description | Supported Values |
+| -------- | ---- | ------ | ------- |
+| `buttonLabel` | String  | Specifies the button label for the `submit` button at the bottom of the enrollment form. Defaults to `Submit`. |
+| `options`     | [UI Schema Element Options ](#ui-schema-element-options-object) |  |
```
Comment (brentschaus-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:710]: Needs description.

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md (hunk 2 above)
Comment (zeusandrade-okta) [reply packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:686]: Not sure what the ask is here. If it is for `elements` missing it's description, we should point it to the `### UI Schema Element object` section below I believe.

Comment (brentschaus-okta) [reply packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:686]: Ah. Thank you!

**Review by zeusandrade-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 4]
```diff
@@ -0,0 +1,748 @@
...
+#### UI Schema Request example
+
+### UI Schema Response object
+
+#### UI Schema Response object properties
+
+| Property      | Type                                                                    | Description                                                                                                       |
+| ------------- | ----------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------- |
+| `id`          | String                                                                  | The unique ID of the UI schema |
+| `uiSchema`    | See [UI Schema Object](#ui-schema-object). | N/A |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:656]:
```suggestion
| `uiSchema`    | See [UI Schema Object](#ui-schema-object) | N/A |
```

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 5]
```diff
@@ -0,0 +1,748 @@
...
+| `lastUpdated` | String (ISO-8601) | Timestamp when the UI Schema was last modified |
+| `_links`      | Object ([JSON HAL](http://tools.ietf.org/html/draft-kelly-json-hal-06)) | Discoverable resources related to the UISchema |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:658]: is it UI Schema or is it `uiSchema`?
```suggestion
| `_links`      | Object ([JSON HAL](http://tools.ietf.org/html/draft-kelly-json-hal-06)) | Discoverable resources related to the UI Schema |
```

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 6]
```diff
@@ -0,0 +1,748 @@
...
+}
+```
+
+### UI Schema object
+
+#### UI Schema properties
+
+| Property      | Type   | Description |
+| ------------- | ------ | ----------- |
+| `buttonLabel` | String | Specifies the button label for the `submit` button at the bottom of the enrollment form. Defaults to `Submit`. |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:685]:
```suggestion
| `buttonLabel` | String | Specifies the button label for the `Submit` button at the bottom of the enrollment form. Defaults to `Submit`. |
```

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 7]
```diff
@@ -0,0 +1,748 @@
...
+| `elements`    | Array  | See [UI Schema Element object](#ui-schema-element-object). |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:686]:
```suggestion
| `elements`    | Array  | See [UI Schema Element object](#ui-schema-element-object) |
```

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 8]
```diff
@@ -0,0 +1,748 @@
...
+| `type`        | String | Type of layout  |
+
+#### UI Schema example
+
+```json
+{
+  "buttonLabel": "Submit",
+  "elements": [],
+  "label": "Sign in",
+  "type": "Group
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:697]:
```suggestion
  "type": "Group"
```

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 9]
```diff
@@ -0,0 +1,748 @@
...
+
+### UI Schema Element object
+
+The UI Schema Element object specifies the configuration of an input field on an enrollment form.
+
+#### UI Schema Element properties
+
+| Property | Type | Description | Supported Values |
+| -------- | ---- | ------ | ------- |
+| `buttonLabel` | String  | Specifies the button label for the `submit` button at the bottom of the enrollment form. Defaults to `Submit`. | N/A |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:709]:
```suggestion
| `buttonLabel` | String  | Specifies the button label for the `Submit` button at the bottom of the enrollment form. Defaults to `Submit`. | N/A |
```

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 10]
```diff
@@ -0,0 +1,748 @@
...
+| `options`     | [UI Schema Element Options ](#ui-schema-element-options-object) | N/A | N/A |
+| `scope`       | String | Specifies the property to which the input field should be bound. Must follow the format `#/properties/PROPERTY_NAME` where `PROPERTY_NAME` is a variable name for an attribute in `profile editor`. | N/A              |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:711]:
```suggestion
| `scope`       | String | Specifies the property to which the input field should be bound. It must follow the format `#/properties/PROPERTY_NAME` where `PROPERTY_NAME` is a variable name for an attribute in `profile editor`. | N/A              |
```

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 11]
```diff
@@ -0,0 +1,748 @@
...
+  "label": "Primary email",
+  "options": {},
+  "scope": "#/properties/email",
+  "type": "Control"
+}
+```
+
+### UI Schema Element Options object
+
+The UI Schema Element Options object specifies how the input is displayed. The supported formats are:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:727]:
```suggestion
The UI Schema Element Options object specifies how the input appears. The supported formats are:
```

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 12]
```diff
@@ -0,0 +1,748 @@
...
+
+- `text`: Text input. This is the default format for the majority of property types.
+- `radio`: Radio button options. This option is only available for `string` data types with an `enum` or `one of` constraint.
+- `select`: Displays input as a dropdown. This option is only available for the `country-code` data type or a `string` data type with an `enum` or `one of` constraint.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:731]:
```suggestion
- `select`: Displays input as a dropdown list. This option is only available for the `country-code` data type or a `string` data type with an `enum` or `one of` constraint.
```

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 13]
```diff
@@ -0,0 +1,748 @@
...
+- `checkbox`: Displays input as a checkbox. This option is only available for `boolean` data types.
+- `radio_yes_no`: Displays input as 2 radio buttons, one with the option `yes` and the other `no`. This option is only available for `boolean` data types.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:733]:
```suggestion
- `radio_yes_no`: Displays input as two radio buttons, one with the option `yes` and the other `no`. This option is only available for `boolean` data types.
```

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 14]
```diff
@@ -0,0 +1,748 @@
...
+- `radio_true_false`: Displays input as 2 radio buttons, one with the option `true` and the other `false`. This option is only available for `boolean` data types.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:734]:
```suggestion
- `radio_true_false`: Displays input as two radio buttons, one with the option `true` and the other `false`. This option is only available for `boolean` data types.
```

File: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md [hunk 15]
```diff
@@ -0,0 +1,748 @@
...
+
+#### UI Schema Element Option properties
+
+| Property | Type   | Description                                 | Supported Values                                                          |
+| -------- | ------ | ------------------------------------------- | ------------------------------------------------------------------------- |
+| `format` | String | Specifies how the input should be displayed | `text`, `radio`, `select`, `checkbox`, `radio_yes_no`, `radio_true_false` |
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/uischema/index.md:740]:
```suggestion
| `format` | String | Specifies how the input should appear | `text`, `radio`, `select`, `checkbox`, `radio_yes_no`, `radio_true_false` |
```

**Review by susanharper-okta — COMMENTED** [review]
couple nits - one question.

**Review by susanharper-okta — APPROVED** [review]
LGTM

**Pull request merged**